modules/SMSNotifier/providers/RingCentral.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * RINGCENTRAL - sms provider.
+ */
+class SMSNotifier_RingCentral_Provider extends SMSNotifier_Basic_Provider
+{
+    /**
+     * Provider name.
+     *
+     * @var string
+     */
+    protected $name = 'RingCentral';
+
+    /**
+     * Address URL.
+     *
+     * @var string
+     */
+    protected $url = 'https://platform.ringcentral.com';
+
+    /**
+     * Encoding.
+     *
+     * @var string
+     */
+    public $encoding = 'utf-8';
+
+    /**
+     * Format.
+     *
+     * @var string
+     */
+    public $format = 'json';
+
+    /**
+     * Required fields.
+     *
+     * @return string[]
+     */
+    public function getRequiredParams()
+    {
+        return ['CLIENT_ID','RINGUSER','RINGPASS','RINGPHONE','RINGEXT'];
+    }
+
+    public function get($key)
+    {
+        return $this->$key;
+    }
+
+    /**
+     * Response.
+     *
+     * @param Requests_Response $request
+     *
+     * @return bool
+     */
+    public function getResponse(Requests_Response $request)
+    {
+        $response = \App\Json::decode($request->body);
+
+        return isset($response['error']) && !empty($response['error']) ? false : true;
+    }
+
+    /**
+     * Fields to edit in settings.
+     *
+     * @return \Settings_Vtiger_Field_Model[]
+     */
+    public function getSettingsEditFieldsModel()
+    {
+        $fields = [];
+        $moduleName = 'Settings:SMSNotifier';
+        foreach ($this->getRequiredParams() as $name) {
+            $field = ['uitype' => 1, 'column' => $name, 'name' => $name, 'displaytype' => 1, 'typeofdata' => 'V~M', 'presence' => 0, 'isEditableReadOnly' => false];
+            if ($name === 'CLIENT_ID') {
+                $field['text'] = [''];
+                $field['label'] = 'CLIENT_ID';
+                $fields[] = $field;
+            }
+            if ($name === 'RINGUSER') {
+                $field['text'] = [''];
+                $field['label'] = 'RINGUSER';
+                $fields[] = $field;
+            }
+            if ($name === 'RINGPASS') {
+                $field['text'] = [''];
+                $field['label'] = 'RINGPASS';
+                $fields[] = $field;
+            }
+            if ($name === 'RINGPHONE') {
+                $field['text'] = [''];
+                $field['label'] = 'RINGPHONE';
+                $fields[] = $field;
+            }
+            if ($name === 'RINGEXT') {
+                $field['text'] = [''];
+                $field['label'] = 'RINGEXT';
+                $fields[] = $field;
+            }
```

Comment:
Good job!
I think you can change this with the following:

```
            if ($name === 'CLIENT_ID') {
                $field['text'] = [''];
                $field['label'] = 'CLIENT_ID';
                $fields[] = $field;
            }
            elseif ($name === 'RINGUSER') {
                $field['text'] = [''];
                $field['label'] = 'RINGUSER';
                $fields[] = $field;
            }
            elseif ($name === 'RINGPASS') {
                $field['text'] = [''];
                $field['label'] = 'RINGPASS';
                $fields[] = $field;
            }
            elseif ($name === 'RINGPHONE') {
                $field['text'] = [''];
                $field['label'] = 'RINGPHONE';
                $fields[] = $field;
            }
            elseif ($name === 'RINGEXT') {
                $field['text'] = [''];
                $field['label'] = 'RINGEXT';
                $fields[] = $field;
            }
```

or with this:

```
            if ($name === 'CLIENT_ID') {
                $field['text'] = [''];
                $field['label'] = 'CLIENT_ID';
                $fields[] = $field;
                continue;
            }
            if ($name === 'RINGUSER') {
                $field['text'] = [''];
                $field['label'] = 'RINGUSER';
                $fields[] = $field;
                continue;
            }
            if ($name === 'RINGPASS') {
                $field['text'] = [''];
                $field['label'] = 'RINGPASS';
                $fields[] = $field;
                continue;
            }
            if ($name === 'RINGPHONE') {
                $field['text'] = [''];
                $field['label'] = 'RINGPHONE';
                $fields[] = $field;
                continue;
            }
            if ($name === 'RINGEXT') {
                $field['text'] = [''];
                $field['label'] = 'RINGEXT';
                $fields[] = $field;
                continue;
            }
```
just for optimization 😃 

Also, you can put `$field['text'] = [''];` and `$fields[] = $field;` before the if statements if is same everytime.

Reply:
a switch would be better